avalon/nuke/pipeline.py

```diff
@@ -215,6 +220,16 @@ def find_host_config(config):
     return config
 
 
+def get_main_window():
+    """Acquire Nuke's main window"""
+    if self._parent is None:
+        top_widgets = QtWidgets.QApplication.topLevelWidgets()
+        main_window = next(widget for widget in top_widgets if
+                           widget.isWindow() and not widget.isHidden())
+        self._parent = main_window
+    return self._parent
```

Comment:
This was bold, but there was no obvious `QWidget` object name in Nuke. :(

Reply:
I did find [this](https://www.mail-archive.com/nuke-python@support.thefoundry.co.uk/msg04726.html) which could be refactored to this:
```python
top_widgets = QtWidgets.QApplication.topLevelWidgets()
name  = "Foundry::UI::DockMainWindow"
main_window = next(widget for widget in top_widgets 
                   if widget.inherits('QMainWindow') and
                   if widget.metaObject().className() == name)
```

Not sure if that works though.

Reply:
Awesome ! This works !!